packages/expo-cli/src/commands/install.ts

```diff
@@ -35,21 +35,33 @@ async function installAsync(packages: string[], options: PackageManager.CreateFo
     log,
   });
 
-  // This ends up being confusing for people. If they're using expo install,
-  // let's just install the deps such that they work in the client even if
-  // it's a bare project.
+  const { exp, pkg } = ConfigUtils.getConfig(projectRoot, { skipSDKVersionRequirement: true });
+
+  // If using `expo install` in a project without the expo package even listed
+  // in package.json, just fall through to npm/yarn.
   //
-  // if (workflow === 'bare') {
-  //   return await packageManager.addAsync(...packages);
-  // }
+  if (!pkg.dependencies['expo']) {
+    return await packageManager.addAsync(...packages);
+  }
+
+  if (!exp.sdkVersion) {
```

Comment:
this has the side effect of improving the messaging in the weird edge case where maybe somebody 1) has expo pkg in pkg.json 2) has node_modules directory but 3) somehow we can't resolve the sdk version